tests/conftest.py

```diff
@@ -11,17 +9,62 @@ def runner(request):
     return CliRunner()
 
 
-def _check_symlinks_supported():
-    with tempfile.TemporaryDirectory(prefix="click-pytest-") as tempdir:
-        target = os.path.join(tempdir, "target")
-        open(target, "w").close()
-        link = os.path.join(tempdir, "link")
+@pytest.fixture(name="assert_no_surrogates", scope="session")
+def assert_no_surrogates_fixture():
+    """This fixture returns a function that can be used to assert that a string
+    can be printed to a stream in which errors='strict' (e.g. stdout), i.e. has no
+    surrogates.
+
+    You *must not remove this from tests*, as many (most?) systems in which our tests
+    will run are already configured with a locale which makes printing surrogates work
+    fine.
+
+    Here are some typical scenarios in which strings with surrogates wouldn't cause an
```

Comment:
I'm not sure why we'd want to optimise for comment length.

The reason for all the information is that unless you have all the locale PEPs in mind (I sure didn't), it's easy to not realise why format_filename was written the way it was prior to click 8.0. I want to avoid the case where someone changes it in the future, and because it works on their machine, thinks that this assertion is unnecessary.